ios/browser/brave_web_main_parts.mm

```diff
@@ -0,0 +1,146 @@
+#include "ios/chrome/browser/browser_state/browser_state_keyed_service_factories.h"
+#include "ios/chrome/browser/browser_state/chrome_browser_state.h"
+#include "ios/chrome/browser/browser_state/chrome_browser_state_manager.h"
+#include "ios/chrome/browser/chrome_paths.h"
+#include "ios/chrome/browser/flags/about_flags.h"
+#include "ios/web/public/thread/web_task_traits.h"
+#include "ios/web/public/thread/web_thread.h"
+#include "ui/base/l10n/l10n_util_mac.h"
+#include "ui/base/resource/resource_bundle.h"
+
+//#if BUILDFLAG(USE_ALLOCATOR_SHIM)
+//#include "base/allocator/allocator_shim.h"
+//#endif
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+BraveWebMainParts::BraveWebMainParts() {}
+
+BraveWebMainParts::~BraveWebMainParts() {}
+
+void BraveWebMainParts::PreEarlyInitialization() {
+  // Uses Private API _performMemoryWarning causing iOS to get rejected.
+  // Crashes very badly on iOS 13.4.0, 13.5.0 according to:
+  // ios_chrome_main_parts.mm -> InitializeAllocatorShim & ShouldInstallAllocatorShim
+//#if BUILDFLAG(USE_ALLOCATOR_SHIM)
+//  base::allocator::InitializeAllocatorShim();
+//#endif
+}
+
+void BraveWebMainParts::PreMainMessageLoopStart() {
+  l10n_util::OverrideLocaleWithCocoaLocale();
+
+   const std::string loaded_locale =
+       ui::ResourceBundle::InitSharedInstanceWithLocale(
+           std::string(), nullptr, ui::ResourceBundle::LOAD_COMMON_RESOURCES);
+   CHECK(!loaded_locale.empty());
+
+   base::FilePath resources_pack_path;
+   base::PathService::Get(ios::FILE_RESOURCES_PACK, &resources_pack_path);
+   ui::ResourceBundle::GetSharedInstance().AddDataPackFromPath(
+       resources_pack_path, ui::SCALE_FACTOR_100P);
+}
+
+void BraveWebMainParts::PreCreateThreads() {
+  scoped_refptr<base::SequencedTaskRunner> local_state_task_runner =
+      base::ThreadPool::CreateSequencedTaskRunner(
+          {base::MayBlock(), base::TaskPriority::BEST_EFFORT,
+           base::TaskShutdownBehavior::BLOCK_SHUTDOWN});
+
+  base::FilePath local_state_path;
+  CHECK(base::PathService::Get(ios::FILE_LOCAL_STATE, &local_state_path));
+  application_context_.reset(new ApplicationContextImpl(
+      local_state_task_runner.get(), *base::CommandLine::ForCurrentProcess(),
+      l10n_util::GetLocaleOverride()));
+  DCHECK_EQ(application_context_.get(), GetApplicationContext());
+
+  // Initialize local state.
+  local_state_ = application_context_->GetLocalState();
+  DCHECK(local_state_);
+
+  flags_ui::PrefServiceFlagsStorage flags_storage(
+      application_context_->GetLocalState());
+  ConvertFlagsToSwitches(&flags_storage,
+                         base::CommandLine::ForCurrentProcess());
+
+  SetupFieldTrials();
+
+  // variations::InitCrashKeys();
+  // metrics::EnableExpiryChecker(::kExpiredHistogramsHashes,
+  //                              ::kNumExpiredHistograms);
+
+  application_context_->PreCreateThreads();
+}
+
+void BraveWebMainParts::SetupFieldTrials() {
```

Comment:
i don't quite understand what this code is doing, but FYI we generally disable field trials: https://github.com/brave/brave-browser/issues/4283

cc @jumde

Reply:
In the official brave browser desktop & ios build command line, I see `fieldtrial_testing_like_official_build=true`